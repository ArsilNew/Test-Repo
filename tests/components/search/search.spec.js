import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";

import { shallow } from "../Utils";
import Search from "../../../src/components/search/search";

const event = { target: { value: "new value" } };

describe("Search", () => {
  test("default snapshot", () => {
    const { tree } = shallow(<Search />);
    expect(tree).toMatchSnapshot();
  });

  test("snapshot with placeholder", () => {
    const { tree } = shallow(
      <Search searchPlaceholder="new placeholder text" />
    );
    expect(tree).toMatchSnapshot();
  });

  test("snapshot with custom search icon", () => {
    const Icon = () => <div>Some Icon</div>;
    const { tree } = shallow(<Search searchIcon={Icon} />);
    expect(tree).toMatchSnapshot();
  });

  test("typing will trigger onChange", () => {
    const onChange = jest.fn();
    const { container } = render(<Search onChange={onChange} />);
    fireEvent.change(container.querySelector("input"), event);
    expect(onChange).toHaveBeenCalledTimes(1);
  });
});
