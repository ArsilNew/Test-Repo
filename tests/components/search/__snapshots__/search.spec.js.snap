// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search default snapshot 1`] = `
<div
  className="search"
>
  <input
    className="input"
    onChange={undefined}
    placeholder="Search..."
    type="text"
  />
  <div
    className="icon"
  >
    <MdSearch />
  </div>
</div>
`;

exports[`Search snapshot with custom search icon 1`] = `
<div
  className="search"
>
  <input
    className="input"
    onChange={undefined}
    placeholder="Search..."
    type="text"
  />
  <div
    className="icon"
  >
    <Icon />
  </div>
</div>
`;

exports[`Search snapshot with placeholder 1`] = `
<div
  className="search"
>
  <input
    className="input"
    onChange={undefined}
    placeholder="new placeholder text"
    type="text"
  />
  <div
    className="icon"
  >
    <MdSearch />
  </div>
</div>
`;
